vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/payment/post_order/post_order.h

```diff
@@ -72,24 +73,24 @@ class PostOrder {
   ~PostOrder();
 
   void Request(
-      const std::vector<type::SKUOrderItem>& items,
+      std::vector<type::SKUOrderItemPtr> items,
       PostOrderCallback callback);
 
  private:
   std::string GetUrl();
 
-  std::string GeneratePayload(const std::vector<type::SKUOrderItem>& items);
+  std::string GeneratePayload(std::vector<type::SKUOrderItemPtr> items);
 
   type::Result CheckStatusCode(const int status_code);
 
   type::Result ParseBody(
       const std::string& body,
-      const std::vector<type::SKUOrderItem>& order_items,
+      std::vector<type::SKUOrderItemPtr> order_items,
       type::SKUOrder* order);
 
   void OnRequest(
       const type::UrlResponse& response,
-      const std::vector<type::SKUOrderItem>& items,
+      std::shared_ptr<std::vector<type::SKUOrderItemPtr>> items,
```

Comment:
why do we need `shared_ptr` here?

Reply:
`shared_ptr` of items need to be passed to the `OnRequest` callback

Reply:
That really doesn't answer the question. Why is a shared_ptr required here? I'm not seeing any reason for it and shared_ptr should generally be avoided